File: dingtalk-mobile-client/src/services/task/TaskOperateLogService.js
import { stringify } from 'qs';
import request from './../../utils/HttpInvoker';

async function findTaskOperateLogList(params) {
  return request(`/api/tm/task-operate-log/list?${stringify(params)}`, {
    method: 'GET',
  });
}

async function findTaskOperateLogWithTaskList(params) {
  return request(`/api/tm/task-operate-log/detail/list?${stringify(params)}`, {
    method: 'GET',
  });
}

export default {
  findTaskOperateLogList,
  findTaskOperateLogWithTaskList
}

File: task-api/src/main/java/com/hozensoft/task/core/service/TaskOperateLogService.java
package com.hozensoft.task.core.service;

import com.github.pagehelper.PageInfo;
import com.hozensoft.struct.pagehelper.Limitable;
import com.hozensoft.struct.pagehelper.Pageable;
import com.hozensoft.system.utils.bean.ContextHolder;
import com.hozensoft.task.core.domain.TaskOperateLog;
import com.hozensoft.task.core.dto.TaskOperateLogItemDto;
import com.hozensoft.task.core.dto.TaskOperateLogQueryDto;
import com.hozensoft.task.core.dto.TaskOperateLogWithTaskItemDto;

import java.util.List;

public interface TaskOperateLogService {

    void addTaskOperateLog(ContextHolder holder, TaskOperateLog taskOperateLog);

    List<TaskOperateLogItemDto> findTaskOperateLogList(ContextHolder holder,
                                                       TaskOperateLogQueryDto params,
                                                       Limitable limitable);

    PageInfo<TaskOperateLogItemDto> findTaskOperateLogPage(ContextHolder holder,
                                                           TaskOperateLogQueryDto params,
                                                           Pageable pageable);

    PageInfo<TaskOperateLogWithTaskItemDto> findTaskOperateLogWithTaskList(ContextHolder holder,
                                                                       Boolean readed,
                                                                       Pageable pageable);
}
